Tensile/Source/client/source/HardwareMonitor.cpp

```diff
@@ -387,9 +391,129 @@ namespace Tensile
                     m_fanValues[i] += newValue;
             }
 
+            rsmi_gpu_metrics_t gpuMetrics;
+            auto status = rsmi_dev_gpu_metrics_info_get(m_smiDeviceIndex, &gpuMetrics);
+            if(status != RSMI_STATUS_SUCCESS)
+            {
+                m_hasInvalidGpuMetricStatus = true;
+            }
+            else
+            {
+                if(!m_hasInvalidGpuMetricStatus)
+                {
+                    m_freqValues.push_back(gpuMetrics.average_gfxclk_frequency);
+                    m_powerValues.push_back(gpuMetrics.average_socket_power);
+                    m_tempHotspotValues.push_back(gpuMetrics.temperature_hotspot);
+                }
+            }
             m_dataPoints++;
         }
 
+        double HardwareMonitor::getAverageMedianGfxFreqPowerTemperature(int freqPowerTempIndex)
+        {
+            assertNotActive();
+
+            if(m_dataPoints == 0)
+                throw std::runtime_error("No data points collected!");
+            if(m_hasInvalidGpuMetricStatus)
+                return std::numeric_limits<double>::quiet_NaN();
+
+            switch(freqPowerTempIndex)
+            {
+            case GPU_FREQUENCY_AVG:
+                return getAverageOrMedianValue(1, m_freqValues); // 1-> Average
+            case GPU_FREQUENCY_MEDIAN:
+                return getAverageOrMedianValue(0, m_freqValues); // 0-> Median
+            case GPU_POWER_AVG:
+                return getAverageOrMedianValue(1, m_powerValues);
+            case GPU_POWER_MEDIAN:
+                return getAverageOrMedianValue(0, m_powerValues);
+            case GPU_TEMPERATURE_AVG:
+                return getAverageOrMedianValue(1, m_tempHotspotValues);
+            case GPU_TEMPERATURE_MEDIAN:
+                return getAverageOrMedianValue(0, m_tempHotspotValues);
+            default:
+                std::cerr << "Invalid statistical type index.\n";
+                return std::numeric_limits<double>::quiet_NaN();
+            }
+        }
+
+        double
+            HardwareMonitor::getAverageOrMedianValue(bool                   isAvg,
```

Comment:
Probably, we can unify getAverageOrMedianValue and getAverageMedianGfxFreqPowerTemperature.
Like 
(1) ave function
  getAverageGfxFreqPowerTemperature(std::vector<uint16_t>& dataValues)
(2) median function
  getMedianGfxFreqPowerTemperature(std::vector<uint16_t>& dataValues)

We can directly calculate the median or avg instead of putting a switch case.